src/server/middlewares/validateVote.js

```diff
@@ -1,175 +1,260 @@
-    where: { sender: vote.sender, festivalQuestion: vote.festivalQuestion },
+    where: {
+      senderAddress,
+      festivalQuestionAddress,
+    },
   });
+
   const localHasVotedArtwork = await Vote.findOne({
-    where: { sender: vote.sender, artworkQuestion: vote.artworkQuestion },
+    where: {
+      senderAddress,
+      artworkQuestionAddress,
+    },
   });
+
   if (localHasVotedFestival || localHasVotedArtwork) {
-    throw Error('has voted');
+    throw Error('Duplicate vote in database');
   }
-};
+}
 
-const checkArtworkQuestionIsHighestVoted = async (vote) => {
-  let highestVote = Array.from(new Set(vote.festivalVoteTokens));
-  highestVote = highestVote.sort((itemA, itemB) => itemB - itemA)[0];
+async function checkArtworkQuestionIsHighestVoted(vote) {
+  // Question does not exist or is not for artworks
   const question = await Question.findOne({
-    where: { address: vote.artworkQuestion },
+    where: {
+      address: vote.artworkQuestionAddress,
+    },
   });
+
   if (!question || !question.artworkId) {
-    throw Error('invalid artwork-question');
+    throw Error('Question is not for artwork answers');
   }
-  const justAnswerIds = vote.festivalAnswers.map((a) => a.id);
-  const indexOfArtwork = justAnswerIds.indexOf(question.artworkId);
-  if (vote.festivalVoteTokens[indexOfArtwork] !== highestVote) {
-    throw Error('invalid artwork-question');
+
+  // We can only vote on properties of an artwork with the highest score of the
+  // previous vote
+  const highestAnswer = vote.festivalVoteTokens
+    .map((tokens, index) => {
+      return {
+        tokens,
+        index,
+      };
+    })
+    .sort((itemA, itemB) => itemB.tokens - itemA.tokens)[0];
+
+  const highestAnswerId = vote.festivalAnswerIds[highestAnswer.index];
+
+  const answer = await Answer.findOne({
+    where: {
+      id: highestAnswerId,
+    },
+  });
+
+  if (question.artworkId !== answer.artworkId) {
+    throw Error('Artwork is not the strongest festival answer');
   }
-};
+}
 
-const checkQuestions = async (vote) => {
-  const activeFestivalQuestion = await admin.methods
-    .questions(vote.festivalQuestion)
+async function checkQuestions(vote) {
+  const activeFestivalQuestion = await adminContract.methods
+    .questions(vote.festivalQuestionAddress)
     .call();
-  const activeArtworkQuestion = await admin.methods
-    .questions(vote.artworkQuestion)
+
+  const activeArtworkQuestion = await adminContract.methods
+    .questions(vote.artworkQuestionAddress)
     .call();
+
   if (!activeFestivalQuestion || !activeArtworkQuestion) {
-    throw Error('inactive question');
+    throw Error('Inactive question');
   }
-};
+}
 
-const checkFestival = async (vote) => {
-  const valid = await isActiveFestival(vote.festival);
-  if (!valid) {
-    throw Error('invalid festival');
+async function checkFestival(vote, questionContract) {
+  const festivalChainId = await questionContract.methods.festival().call();
```

Comment:
Minor, but the festival id is validated elsewhere, you don't have to call out here